### DISCUSSION TITLE
How to Allow Either JWT or API Key to Pass Authentication

### DISCUSSION THREAD

**Discussion by s35241607** [Kong/kong#14052] [Help] [answered]
Hi everyone,

I'm currently using KONG as my API Gateway, and I have a requirement to allow both JWT and API Key authentication on the same Route, where passing either one would be sufficient. However, I encountered some issues.

Here are the two approaches I've tried so far, but neither worked as expected:

Enabling both JWT and key-auth plugins on the same Route, but this requires both to pass authentication, which is not what I need.
Setting anonymous for both the JWT and key-auth plugins, but this allows requests to pass even when neither JWT nor API Key is provided.
Is there a way to achieve the behavior where either JWT or API Key can individually pass authentication?
Should I adjust my configuration or use a different approach?

Thanks in advance for your help!

**Comment by Oyami-Srk** [comment, marked as answer]
The second approach is correct. According to the [documentation](https://docs.konghq.com/gateway/latest/kong-plugins/authentication/reference/#multiple-authentication), setting anonymous will make a `OR` logic.
You may want to use the [Request Termination](https://docs.konghq.com/hub/kong-inc/request-termination/) plugin to disallow the anonymous user from accessing.

**Reply by s35241607** [reply]
thank you, it works
